zipline/src/engineTest/kotlin/app/cash/zipline/SampleService.kt

```diff
@@ -52,74 +51,55 @@ interface SampleService : ZiplineService {
       : ZiplineServiceAdapter<SampleService>(), KSerializer<SampleService> {
       override val serialName: String = "SampleService"
 
-      override fun inboundCallHandler(
-        service: SampleService,
-        context: InboundBridge.Context
-      ): InboundCallHandler = GeneratedInboundCallHandler(service, context)
-
-      override fun outboundService(
-        context: OutboundBridge.Context
-      ): SampleService = GeneratedOutboundService(context)
-
-      private class GeneratedInboundCallHandler(
-        private val service: SampleService,
-        override val context: InboundBridge.Context,
-      ) : InboundCallHandler {
-        private val requestSerializer = context.serializersModule.serializer<SampleRequest>()
-        private val responseSerializer = context.serializersModule.serializer<SampleResponse>()
-
-        private val supportedFunctionNames = listOf(
-          "fun close(): kotlin.Unit",
-          "fun ping(app.cash.zipline.SampleRequest): app.cash.zipline.SampleResponse",
+      override fun ziplineFunctions(
```

Comment:
# ⚠️

SampleService.kt is a great place to start . . . see what the before/after generated code looks like, and everything else mostly falls out from there.